File: frontend/src/api/quote.ts
import { http } from './http';

export type QuoteLine = { beerId: string; quantity: number };
export type QuoteRequest = { items: QuoteLine[] };

export type QuoteResponse = {
  wholesaler: string;
  totalQuantity: number;
  subtotal: number;
  discountRate: number;
  total: number;
  lines: {
    beerId: string;
    beerName: string;
    quantity: number;
    unitPrice: number;
    lineTotal: number;
  }[];
};

export function requestQuote(wholesalerId: string, payload: QuoteRequest) {
  return http<QuoteResponse>(`/wholesalers/${wholesalerId}/quote`, {
    method: 'POST',
    body: JSON.stringify(payload),
  });
}

File: backend/Api/Dto/Quote.cs
namespace Api.Dto;

public sealed record QuoteLine(Guid BeerId, int Quantity);

public sealed record QuoteRequest(List<QuoteLine> Items);

public sealed record QuoteLineResult(
	Guid BeerId,
	string BeerName,
	int Quantity,
	decimal UnitPrice,
	decimal LineTotal);

public sealed record QuoteResponse(
	string Wholesaler,
	int TotalQuantity,
	decimal Subtotal,
	decimal DiscountRate,
	decimal Total,
	List<QuoteLineResult> Lines);